src/V3PreLex.l

```diff
@@ -112,12 +117,102 @@ bom             [\357\273\277]
+                          if (res == 0)
+                              LEXP->curFilelinep()->v3warn(PROTECTERR, "`pragma protected encoding must have an \"enctype\" field");
+                          LEXP->m_encType = !strcasecmp(enctype, "uuencode") ? Enctype::UUENCODE :
+                                            !strcasecmp(enctype, "base64") ? Enctype::BASE64 :
+                                            !strcasecmp(enctype, "quoted-printable") ? Enctype::QUOTE_PRINTABLE :
+                                            !strcasecmp(enctype, "raw") ? Enctype::RAW : Enctype::ERR;
+                          if (LEXP->m_encType == Enctype::ERR)
+                              LEXP->curFilelinep()->v3warn(PROTECTERR, "Illegal encoding type for `pragma protected encoding");
+                          if (LEXP->m_encType != Enctype::BASE64)
+                              LEXP->curFilelinep()->v3warn(PROTECTERR, "At the moment, only BASE64 is recognized for `pragma protected encoding");
+                          if (res == 3) {
+                              if ((LEXP->m_encType == Enctype::BASE64) && (LEXP->m_protLength & 3))
+                                  LEXP->curFilelinep()->v3warn(PROTECTERR, "line_length must be multiple of 4 for BASE64");
+                          } else {
+                              // default values
+                              LEXP->m_protBytes = 0;
+                              LEXP->m_protLength = 76; // ?? default value not mentioned in IEEE spec
+                          }
+                          BEGIN(INITIAL);
+                          return VP_TEXT;
+                        }
+<PRAGMAPROTECT>"key_block"{wsn}*[\n\r] {
+                          FL_FWDC; linenoInc();
+                          BEGIN(ENCBASE64);
+                          return VP_TEXT; }
+<PRAGMAPROTECT>"data_block"{wsn}*[\n\r] {
+                          FL_FWDC; linenoInc();
+                          LEXP->curFilelinep()->v3warn(PROTECTED, "A '`pragma protected data_block' encrypted section was detected and will be skipped.");
+                          BEGIN(ENCBASE64);
+                          return VP_TEXT; }
+<PRAGMAPROTECT>("begin_protected"|"end_protected")[\n\r]  { FL_FWDC; linenoInc(); BEGIN(INITIAL); return VP_TEXT; }
+<PRAGMAPROTECT>"version="[^\n\r]*[\n\r] { FL_FWDC; linenoInc(); BEGIN(INITIAL); return VP_TEXT; }
+<PRAGMAPROTECT>("encrypt_agent"|"encrypt_agent_info"|"key_keyowner"|"key_keyname"){wsn}*={wsn}*[\"][^\n\r]*[\"][\n\r] {
+                          FL_FWDC; linenoInc(); BEGIN(INITIAL); return VP_TEXT; }
+<PRAGMAPROTECT>("data_method"|"key_method"){wsn}*={wsn}*[\"][^\n\r]*[\"][\n\r] {
+    FL_FWDC; linenoInc(); BEGIN(INITIAL); return VP_TEXT; }
+    /* catch-all for unknown '`pragma protect' rules */
+<PRAGMAPROTECT>. { yyless(0); BEGIN(PRAGMAPROTERROR); return VP_TEXT; }
```

Comment:
Fix indent please.

Reply:
OK, I'll merge to branch first.

Reply:
I'm fixing the indents but I have to say they are unusual.

All code blocks are deeply indented, but only the rule's outermost curly braces are aligned on a 4-tab stop.
The code inside the rule braces is indented 2 more spaces, to align below any statement right after the opening indent, but any further indents are again 4-spaced, leading to 4N+2 indents.
Some rules have 2 short statements per line
Some (most) rules have the terminating left brace at the end of the last statement.

I understand that this comes from the fact that most lexer patterns are short and that the code following them can fit on the same line, but for longer rules, it is not so good.

Reply:
Yes, it's kind of a historical mess.  Do whatever looks consistent.